app/javascript/v3/components/Form/Button.vue

```diff
@@ -0,0 +1,142 @@
+<template>
+  <button
+    class="inline-flex items-center gap-1 text-sm font-medium reset-base rounded-xl w-fit"
+    :type="type"
+    :class="buttonClasses"
+    :disabled="isDisabled"
+    @click="handleClick"
+  >
+    <fluent-icon
+      v-if="shouldShowLeftIcon"
+      :size="iconSize"
+      :icon="icon"
+      class="flex-shrink-0"
+    />
+    <span
+      v-if="$slots.default && !iconOnly"
+      class="text-sm font-medium text-left truncate rtl:text-right"
+    >
+      <slot />
+    </span>
+    <fluent-icon
+      v-if="shouldShowRightIcon"
+      :size="iconSize"
+      icon="chevron-right"
+      class="flex-shrink-0"
+    />
+  </button>
+</template>
+<script setup>
+import { computed, defineProps } from 'vue';
+
+const props = defineProps({
+  type: {
+    type: String,
+    default: 'submit',
+    validator: value => ['button', 'submit', 'reset'].includes(value),
+  },
+  variant: {
+    type: String,
+    default: '',
+  },
+  size: {
+    type: String,
+    default: '',
+  },
+  icon: {
+    type: String,
+    default: '',
+  },
+  colorScheme: {
+    type: String,
+    default: 'primary',
+  },
+  classNames: {
+    type: [String, Object],
+    default: '',
+  },
```

Comment:
classnames should not be added as a prop, anyway a class applied on the component usage will show up here. We can use the `$attrs` to access the attributes provided to the component in case we need it